homeassistant/util/unit_system.py

```diff
@@ -116,6 +139,41 @@ def as_dict(self) -> dict:
             VOLUME: self.volume_unit
         }
 
+    def convert(self, state, unit_of_measure):
+        """Generic conversion method."""
+        converted = None
+        try:
+            if (unit_of_measure in (TEMP_CELSIUS, TEMP_FAHRENHEIT) and
+                    unit_of_measure != self.temperature_unit):
+                # Convert temperature if we detect one
+                prec = len(state) - state.index('.') - 1 if '.' in state else 0
+                temp = self.temperature(float(state), unit_of_measure)
+                state = str(round(temp) if prec == 0 else round(temp, prec))
+                converted = {}
+                converted["value"] = state
+                converted["units"] = self.temperature_unit
+        except ValueError:
+            # Could not convert state to float
+            pass
```

Comment:
We shouldn't `pass` when an exception happens, no other code will do anything. Just `return None`